src/main/java/me/moodcat/api/RoomAPI.java

```diff
@@ -174,4 +174,28 @@ public NowPlaying getCurrentTime(@PathParam("id") final int roomId) {
         return nowPlaying;
     }
 
+    /**
+     * Process a vote to a song. A vote is either "like" or "dislike".
+     *
+     * @param vote
+     *            The vote.
+     * @return The song object, if the process was succesful.
+     * @throws InvalidVoteException
+     *             Thrown if an invalid vote was cast.
+     */
+    @POST
+    @Path("{id}/vote/{vote}")
+    @Transactional
+    public RoomModel voteSong(@PathParam("id") final int roomId,
+            @PathParam("vote") final String vote) throws InvalidVoteException {
+        final RoomInstance roomInstance = this.backend.getRoomInstance(roomId);
+
+        if ("like".equals(vote) || "dislike".equals(vote)) {
```

Comment:
The default valueOf method of an enum throws an IllegalArgumentException when the value is not in the enum. I dont like the string because it seems magic and there is an enum anyways.